src/index.js

```diff
@@ -15,15 +15,16 @@ export function sampleRUM(checkpoint, data) {
   const timeShift = () => (window.performance ? window.performance.now() : Date.now() - window.hlx.rum.firstReadTime);
   try {
     window.hlx = window.hlx || {};
-    if (!window.hlx.rum) {
+    if (!window.hlx.rum || (window.hlx.rum && typeof window.hlx.rum.collector === 'undefined')) {
       sampleRUM.enhance = () => {};
       const param = new URLSearchParams(window.location.search).get('rum');
       const weight = (param === 'on' && 1)
         || (window.SAMPLE_PAGEVIEWS_AT_RATE === 'high' && 10)
         || (window.SAMPLE_PAGEVIEWS_AT_RATE === 'low' && 1000)
         || 100;
-      const id = crypto.getRandomValues(new Uint32Array(1))[0].toString(36).slice(-4);
-      const isSelected = (param !== 'off') && (Math.random() * weight < 1);
+      const id = (window.hlx.rum && window.hlx.rum.id)
+        || crypto.randomUUID().slice(-9);
+      const isSelected = (window.hlx.rum && window.hlx.rum.isSelected) || ((param !== 'off') && (Math.random() * weight < 1));
```

Comment:
Isnt this evaluating Math.random() even if the microtel script has benn run before? (isSelected is not set in that case)